### ISSUE TITLE
Dexcom Share CGM broken by changes in data format for OUS accounts

### ISSUE THREAD

**Issue by bjornoleh** [LoopKit/Loop#1576]
The Dexcom Share CGM for Loop (https://github.com/LoopKit/dexcom-share-client-swift) recently got broken by changes in the data format for "Outside of the US" (OUS) accounts. It is likely US accounts will be affected in the future.

More info can be found in 
https://github.com/nightscout/share2nightscout-bridge/issues/52

and hints of the solution can be found in 

https://github.com/nightscout/share2nightscout-bridge/commit/976fce416140d9b4b1e43d3704d4c0778ab57064

and Nightscout commits on Dec 2, 2021

https://github.com/nightscout/cgm-remote-monitor/commits/8d83617c92c2b43f9c3ab03d653404e6fa7c325c

**To Reproduce**

Enter Dexcom Share credentials for Share CGM
This is the result, failing at interpreting trend data:

<img src="https://user-images.githubusercontent.com/63544115/144724108-0a9d9f3a-044e-47a7-aa15-d4635eb96a99.jpeg" width="300">

**Phone**
iPhone 13 mini, iOS 15.1

**Loop Version**
 - Version Number: 2.2.6

**CGM**
Dexcom G6
Outside of US Dexcom account
Old style username (not email address)

**Comment by codelinx** [comment]
just received this error also and i cannot get any details in the loop app.
![image](https://user-images.githubusercontent.com/5210394/145340827-4258c0c0-a95e-4e23-a085-c32ae9aae785.png)

**Comment by nicnac88** [comment]
Following. We are experiencing the same issue and getting the same error. I found an Aug 2019 posting that suggested removing the share app credentials  from loop would resolve, but it hasn’t worked for us thus far.

**Comment by dybs** [comment]
This post in the Looped FB group talks about this issue and also suggests removing the share credentials.
https://www.facebook.com/groups/TheLoopedGroup/permalink/2999936863556293

**Comment by codelinx** [comment]
> Following. We are experiencing the same issue and getting the same error. I found an Aug 2019 posting that suggested removing the share app credentials  from loop would resolve, but it hasn’t worked for us thus far. 

Based on issue report from the app "failed  to decode sgv record"

````

## LoopVersion
* Version: Loop (Automatic Bolusing) v2.2.4
* gitRevision: 0e51062f5f5c2e12a3e7d3c15808313950cd8a5a
* gitBranch: (HEAD detached at 0e51062f)
* sourceRoot: /Users/.../Downloads/BuildLoop/Loop-AutomaticBolus-210623-1657/LoopWorkspace/Loop
* buildDateString: Thu Jun 24 00:55:55 EDT 2021
* xcodeVersion: 12D4e

## FeatureFlags
FeatureFlagConfiguration(sensitivityOverridesEnabled: true)

## DeviceDataManager
* launchDate: 2021-12-09 05:02:48 +0000
* lastError: Optional((date: 2021-12-09 08:53:21 +0000, error: ShareClient.ShareError.dataError(reason: "Failed to decode an SGV record: [{\"WT\":\"Date(1639039953000)\",\"ST\":\"Date(1639039953000)\",\"DT\":\"Date(1639039953000-0500)\",\"Value\":48,\"Trend\":\"Flat\"},{\"WT\":\"Date(1639039654000)\",\"ST\":\"Date(1639039654000)\",\"DT\":\"Date(1639039654000-0500)\",\"Value\":47,\"Trend\":\"FortyFiveDown\"},{\"WT\":\"Date(1639039354000)\",\"ST\":\"Date(1639039354000)\",\"DT\":\"Date(1639039354000-0500)\",\"Value\":44,\"Trend\":\"SingleDown\"},{\"WT\":\"Date(1639039053000)\",\"ST\":\"Date(1639039053000)\",\"DT\":\"Date(1639039053000-0500)\",\"Value\":55,\"Trend\":\"FortyFiveDown\"},{\"WT\":\"Date(1639038754000)\",\"ST\":\"Date(1639038754000)\",\"DT\":\"Date(1639038754000-0500)\",\"Value\":72,\"Trend\":\"FortyFiveDown\"},{\"WT\":\"Date(1639038454000)\",\"ST\":\"Date(1639038454000)\",\"DT\":\"Date(1639038454000-0500)\",\"Value\":80,\"Trend\":\"Flat\"}]")))
* lastBLEDrivenUpdate: 2021-12-09 08:53:20 +0000

## G6CGMManager
latestReading: nil
latestConnection: nil
dataIsFresh: false
providesBLEHeartbeat: false
## ShareClientManager
latestBackfill: nil

observers.count: 0
## Transmitter
## BluetoothManager
No peripheral

## OmnipodPumpManager
podComms: ## PodComms
podState: Optional(### PodState
* address: xxxxx
* setupUnitsDelivered: Optional(2.8)
* piVersion: 2.10.0
* pmVersion: 2.10.0
* lot: 71204
* tid: 2500413
* suspendState: resumed(2021-12-08 20:21:24 +0000)
* unfinalizedBolus: nil
* unfinalizedTempBasal: nil
* unfinalizedSuspend: nil
* unfinalizedResume: Optional(Resume: 12/5/21, 11:27:54 PM Certain)
* finalizedDoses: []
* activeAlerts: [OmniKit.AlertSlot.slot7: Pod expiration advisory alarm]
* messageTransportState: MessageTransportState(packetNumber: 15, messageNumber: 9)
* setupProgress: completed
* primeFinishTime: Optional(2021-12-06 04:26:20 +0000)
* configuredAlerts: [OmniKit.AlertSlot.slot2: Shutdown imminent alarm, OmniKit.AlertSlot.slot7: Pod expiration advisory alarm]

fault: nil
)
configuredDevices: ["xxxx"]
delegate: true

````

**Comment by marionbarker** [comment]
A proposed hotfix branch is out for testing for FreeAPS.
* https://github.com/loopnlearn/LoopWorkspace/issues/18

**Comment by marionbarker** [comment]
@hpc4fun 
I tested your solution - much more elegant.  I have updated the hotfix/dexcom-share branch of dexcom-share-client-swift in the loopnlearn repo.

I was a little uncertain of the actual strings for trendmap positions 8 and 9 - they disagree with the share2bridge javascript, so I put in a nonsense string instead of my current arrow string and appropriately got no arrow returned.
I also put in the old `Trend as int code` as an `else if` before the throwError. I can't test this, but if Dexcom reverts this change, this should work as before.

**Comment by codelinx** [comment]
Should this be closed since the change was accepted into a working branch?

**Comment by bjornoleh** [comment]
> Should this be closed since the change was accepted into a working branch?

No, it’s not fixed in LoopKit/Loop yet, so better leave it open.

**Comment by marionbarker** [comment]
This has been fixed in Loop dev branch, dexcom-share-client-swift commit 1faf69f. I do not think any updates to Loop master are planned.
Not sure this should be kept open.

**Comment by bjornoleh** [comment]
There has been mention of adding Dash support to the Loop 2.x line when it is finalised in dev and Loop 3.0 is released. Meaning that there will eventually be a new (final?) release of Loop2.x. Perhaps keep this issue open so it can be fixed for Loop 2.x as well when this happens?

**Comment by bjornoleh** [comment]
Looks like this is fixed in Loop master as well with release 2.2.9

https://github.com/LoopKit/dexcom-share-client-swift/commit/b0a6bd2e649f900b5928358662d1c77985ccaec3
